internal/client/apiclient.go

```diff
@@ -47,18 +47,39 @@ func NewAPIClient(ctx context.Context, addr string, authcb func(string), options
 
 	apiClient := public.NewAPIClient(clientConfig)
 
+	startTime := time.Now()
 	resp, _, err := apiClient.AuthApi.DeviceStart(ctx).Execute()
 	if err != nil {
 		return nil, err
 	}
 
+	// knowing the right time is needed to validate if the JWT has expired
+	serverTimeFn := time.Now
+
+	// Can we use the server's reported time to detect if our time is skewed?
+	if !resp.ServerTime.IsZero() {
+
+		// account for the time spend sending the request to the server...
+		rtt := time.Since(startTime)
```

Comment:
does it make sense to log a warning about this time skew? because there is a good possibility that the device clock can get skewed out of grace period and never reconcile ?

Reply:
We don't pass a logger instance to the client, so that's why I didn't log it.  But with this PR we use a new Now function implementation that adjusts for the time skew which will allow reconciles to succeed.